engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -964,7 +965,16 @@ else if (allocatedContainers.isEmpty() && numRequestedContainers == 0 && dnmgr.c
         appDone = true;
       }
 
-      LOG.debug("Current application state: loop=" + loopCounter + ", appDone=" + appDone + ", total=" + numTotalContainers + ", requested=" + numRequestedContainers + ", released=" + numReleasedContainers + ", completed=" + numCompletedContainers + ", failed=" + numFailedContainers + ", currentAllocated=" + allocatedContainers.size());
+      if (!StringUtils.isEmpty(dnmgr.shutdownDiagnosticsMessage)) {
```

Comment:
The expectation is that the issue is fixed, so INFO is not needed. The ERROR, WARN and INFO should target application's designers and dev-ops teams. DEBUG is for Apex developers.